agent/engine/docker_task_engine_test.go

```diff
@@ -527,11 +526,17 @@ func TestSteadyStatePoll(t *testing.T) {
 		// the engine *may* call StopContainer even though it's already stopped
 		client.EXPECT().StopContainer("containerId", stopContainerTimeout).AnyTimes(),
 	)
+	wait.Wait()
+
+	cleanupChan := make(chan time.Time)
+	testTime.EXPECT().After(gomock.Any()).Return(cleanupChan).AnyTimes()
+	client.EXPECT().RemoveContainer(gomock.Any(), gomock.Any()).Return(nil)
+	imageManager.EXPECT().RemoveContainerReferenceFromImageState(gomock.Any()).Return(nil)
```

Comment:
The parameter here is the `container` per the [code](https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/docker_image_manager.go#L156), if we want to be more deterministic, there will be extra change and actually this part isn't important in this test, this part just ensure all the go routines has finished. So I think it's fine to leave it as it is.